app/src/main/java/ch/epfl/sweng/erpa/activities/sketches/DieSketch.java

```diff
@@ -118,25 +69,23 @@ public void draw() {
         directionalLight(200, 200, 250, 0, -1, 2);
 
         pushMatrix();
-        translate(250, 250, -50);
+        translate(170, 170, -50);
         scale(-100f);
-        if (iX == 0)
-            staticDie();
-        else
+
+        if (millis() - rollStarted > rollLength && rolling) {
+            setResult();
+            rolling = false;
+        }
+
+        if (rolling) {
+            iZ = iX;
```

Comment:
if iZ and iX are equal, why do you want to have two different variables? (I had choose to use two variables so that the rotation along the x-axis and the z-axis would not have the same speed. You can set the same speed but in that case isn't a single variable enough?)